span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java

```diff
@@ -90,25 +83,16 @@ public static JaegerSpanNormalizer get(Config config) {
 
   public JaegerSpanNormalizer(Config config) {
     try {
-      if (config.hasPath(SpanNormalizerConstants.PII_KEYS_CONFIG_KEY)) {
-        config.getStringList(SpanNormalizerConstants.PII_KEYS_CONFIG_KEY).stream()
-            .map(String::toUpperCase)
-            .forEach(tagKeysToRedact::add);
-        redactedAttributeValue =
-            AttributeValue.newBuilder()
-                .setValue(SpanNormalizerConstants.PII_FIELD_REDACTED_VAL)
-                .build();
-      }
-      if (config.hasPath(SpanNormalizerConstants.PII_REGEX_CONFIG_KEY)) {
-        config.getStringList(SpanNormalizerConstants.PII_REGEX_CONFIG_KEY).stream()
-            .map(Pattern::compile)
-            .forEach(tagRegexPatternToRedact::add);
-        if (redactedAttributeValue == null) {
-          redactedAttributeValue =
-              AttributeValue.newBuilder()
-                  .setValue(SpanNormalizerConstants.PII_FIELD_REDACTED_VAL)
-                  .build();
-        }
+      if (config.hasPath(SpanNormalizerConstants.SPAN_REDACTION_CONFIG_KEY)) {
+        config.getConfigList(SpanNormalizerConstants.PII_PCI_CONFIG_KEY).stream()
+            .map(
+                conf ->
+                    new PIIPCIField(
```

Comment:
Either use chaining or builder to create new objects.

Reply:
I don't think it is required over here. What's wrong with this approach though? 
I see this pattern already being used by traceable team: [https://github.com/razorpay/hypertrace-ingester/blob/7b03729ab4930cabc7525ae97a7312907d9acafc/span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/SpanFilter.java#L80](https://github.com/razorpay/hypertrace-ingester/blob/7b03729ab4930cabc7525ae97a7312907d9acafc/span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/SpanFilter.java#L80)